pkg/registry/apis/folders/legacy_storage.go

```diff
@@ -0,0 +1,265 @@
+		p.Spec.Title = strings.ReplaceAll(p.Spec.Title, "${RAND}", rand)
+	}
+
+	accessor := kinds.MetaAccessor(p)
+	parent := accessor.GetFolder()
+
+	out, err := s.service.Create(ctx, &folder.CreateFolderCommand{
+		SignedInUser: user,
+		UID:          p.Name,
+		Title:        p.Spec.Title,
+		Description:  p.Spec.Description,
+		OrgID:        info.OrgID,
+		ParentUID:    parent,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return s.Get(ctx, out.UID, nil)
+}
+
+func (s *legacyStorage) Update(ctx context.Context,
+	name string,
+	objInfo rest.UpdatedObjectInfo,
+	createValidation rest.ValidateObjectFunc,
+	updateValidation rest.ValidateObjectUpdateFunc,
+	forceAllowCreate bool,
+	options *metav1.UpdateOptions,
+) (runtime.Object, bool, error) {
+	info, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, false, err
+	}
+
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	created := false
+	oldObj, err := s.Get(ctx, name, nil)
+	if err != nil {
+		return oldObj, created, err
+	}
+
+	obj, err := objInfo.UpdatedObject(ctx, oldObj)
+	if err != nil {
+		return oldObj, created, err
+	}
+	f, ok := obj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, created, fmt.Errorf("expected folder after update")
+	}
+	old, ok := oldObj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, created, fmt.Errorf("expected old object to be a folder also")
+	}
+
+	oldParent := kinds.MetaAccessor(old).GetFolder()
+	newParent := kinds.MetaAccessor(f).GetFolder()
+	if oldParent != newParent {
+		_, err = s.service.Move(ctx, &folder.MoveFolderCommand{
+			SignedInUser: user,
+			UID:          name,
+			OrgID:        info.OrgID,
+			NewParentUID: newParent,
+		})
+		if err != nil {
+			return nil, created, fmt.Errorf("error changing parent folder spec")
+		}
+	}
+
+	changed := false
+	cmd := &folder.UpdateFolderCommand{
+		SignedInUser: user,
+		UID:          name,
+		OrgID:        info.OrgID,
+	}
+	if f.Spec.Title != old.Spec.Title {
+		cmd.NewTitle = &f.Spec.Title
+		changed = true
+	}
+	if f.Spec.Description != old.Spec.Description {
+		cmd.NewDescription = &f.Spec.Description
+		changed = true
+	}
+	if changed {
+		_, err = s.service.Update(ctx, cmd)
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	r, err := s.Get(ctx, name, nil)
+	return r, created, err
+}
+
+// GracefulDeleter
+func (s *legacyStorage) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	v, err := s.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return v, false, err // includes the not-found error
+	}
+	info, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, false, err
+	}
+	p, ok := v.(*v0alpha1.Folder)
+	if !ok {
+		return v, false, fmt.Errorf("expected a folder response from Get")
+	}
+	err = s.service.Delete(ctx, &folder.DeleteFolderCommand{
+		UID:   name,
+		OrgID: info.OrgID,
+		// ForceDeleteRules: true, ????
```

Comment:
If the folder contains alert rules then the request will fail unless we set `ForceDeleteRules` to `true`. But I think it's safer not to enable it.